backend/services/deviceconnect/api/http/device.go

```diff
@@ -230,11 +233,14 @@ func (h DeviceController) connectWSWriter(
 		conn.Close()
 	}()
 
-	// handle the ping-pong connection health check
-	if err != nil {
-		l.Error(err)
-		return err
-	}
+	// send periodic ping messages to keep the connection alive
+	pingPeriod := (pongWait * 9) / 10
+	ticker := time.NewTicker(pingPeriod)
+	defer ticker.Stop()
+	conn.SetPongHandler(func(string) error {
+		ticker.Reset(pingPeriod)
+		return conn.SetReadDeadline(time.Now().Add(pongWait))
+	})
 
 	conn.SetPingHandler(func(msg string) error {
```

Comment:
Also on ping we can defer sending ping when the connection is idle.
```suggestion
	conn.SetPingHandler(func(msg string) error {
		ticker.Reset(pingPeriod)
```